### PR TITLE
Nostr market

### PR DESCRIPTION
**Pull request by talvasconcelos**
Nostrifying the Market extension

### CODE REVIEW

File: cbc.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from Cryptodome.Cipher import AES
```
Comment (motorina0) [inline cbc.py]: @talvasconcelos can we remove this file or do you plan to use it somewhere?

File: crud.py [hunk 2]
```diff
@@ -46,6 +48,10 @@ async def create_market_product(data: createProduct) -> Products:
     )
     product = await get_market_product(product_id)
     assert product, "Newly created product couldn't be retrieved"
+    stall = await get_market_stall(product.stall)
```
Comment (motorina0) [inline crud.py:51]: - duplicate logic?
- `get_market_stall()` already does the `product.price` computation

File: crud.py [hunk 3]
```diff
@@ -57,13 +63,28 @@ async def update_market_product(product_id: str, **kwargs) -> Optional[Products]
...
+        product.price /= stall.fiat_base_multiplier
 
-    return Products(**row) if row else None
+    return product
 
 
 async def get_market_product(product_id: str) -> Optional[Products]:
     row = await db.fetchone("SELECT * FROM market.products WHERE id = ?", (product_id,))
-    return Products(**row) if row else None
+    product = Products(**row) if row else None
```
Comment (motorina0) [inline crud.py:79]: - suggestion: introduce `Product.from_row(row, currency, fiat_base_multiplier)`
- have this logic in one place alone

File: crud.py [hunk 4]
```diff
@@ -91,17 +122,19 @@ async def delete_market_product(product_id: str) -> None:
...
-        """
+        f"""
         INSERT INTO market.zones (
             id,
             "user",
             cost,
-            countries
+            countries,
+            currency,
+            stall
```
Comment (motorina0) [inline crud.py:132]: - `zones` should not know about `stalls`

File: crud.py [hunk 5]
```diff
@@ -139,24 +172,26 @@ async def delete_market_zone(zone_id: str) -> None:
...
     await db.execute(
-        """
+        f"""
         INSERT INTO market.stalls (
             id,
             wallet,
             name,
             currency,
             publickey,
+            privatekey,
```
Comment (motorina0) [inline crud.py:182]: - why does a `stall` need a `privatekey`?
- I thought there is one key per merchant

File: crud.py [hunk 6]
```diff
@@ -205,17 +256,18 @@ async def delete_market_stall(stall_id: str) -> None:
 ###Orders
 
 
-async def create_market_order(data: createOrder, invoiceid: str):
+async def create_market_order(data: createOrder, invoiceid: str, order_id: str):
```
Comment (motorina0) [inline crud.py:259]: - nitpick: make `order_id` an Optional field of the `createOrder` class

File: cbc.py (hunk 1 above)
Comment (talvasconcelos) [reply cbc.py]: it can be removed, what was needed it's on helpers.py

File: crud.py (hunk 5 above)
Comment (talvasconcelos) [reply crud.py:182]: we either have it there or we need a global "settings" for merchant with the keypair...

File: crud.py (hunk 4 above)
Comment (talvasconcelos) [reply crud.py:132]: The goal is to have zones per stall. If they are "global", it causes issues with the shipping cost unit (sat, EUR, USD, ...)

**Comment by motorina0** [comment]
@talvasconcelos can this be closed?

**Pull request closed without merging**